src/fpylll/gso.pyx

```diff
@@ -13,6 +13,11 @@ from util cimport preprocess_indices, check_float_type
 from fpylll cimport mpz_double, mpz_ld, mpz_dpe, mpz_mpfr, mpz_dd, mpz_qd
 from fplll cimport FT_DOUBLE, FT_LONG_DOUBLE, FT_DD, FT_QD, FT_DPE, FT_MPFR, FloatType
 
+import cython
+import numpy as np
```

Comment:
>  I presume you need the double* because it is the interface for another C/C++ program, then? 

Yes ;) Python will basically create the object, have fplll fill it up, and then pass it to another C++ program. The object will be reused many times so python overhead won't matter here.

As for the loop in Cython, I am unsure of what you have in mind.